odoo/addons/base/models/ir_model.py

```diff
@@ -937,6 +914,165 @@ def _add_manual_fields(self, model):
                     model._add_field(name, field)
 
 
+class IrModelSelection(models.Model):
+    _name = 'ir.model.fields.selection'
+    _order = 'sequence, id'
+    _description = "Fields Selection"
+
+    field_id = fields.Many2one("ir.model.fields",
+        required=True, ondelete="cascade",
+        domain=[('ttype', 'in', ['selection', 'reference'])])
+    value = fields.Char(required=True)
+    name = fields.Char(translate=True, required=True)
+    sequence = fields.Integer(default=10)
+
+    _sql_constraints = [
+        ('selection_field_uniq', 'unique(field_id, value)',
+         'Selections values must be unique per field'),
+    ]
+
+    @tools.ormcache()
+    def _get_field_data(self, field_id):
+        """ Return the given field's selection. """
+        self._cr.execute("""
+            SELECT value, name
+            FROM ir_model_fields_selection
+            WHERE field_id=%s
+            ORDER BY sequence, id
+        """, (field_id,))
+        return self._cr.fetchall()
+
+    @tools.ormcache('model_name')
+    def _existing_field_data(self, model_name):
```

Comment:
Selection and reference fields are not so common, so trying to query selections by model looks like premature optimization. I suggest to simply take `field` as parameter. This will align the APIs of `_existing_field_data` and `_reflect_field_params`, and make their comparison trivial.